gammapy/datasets/actors.py

```diff
@@ -0,0 +1,188 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import copy
+import inspect
+import logging
+import numpy as np
+from gammapy.modeling.models import DatasetModels
+from .core import Dataset, Datasets
+from .map import MapDataset
+
+log = logging.getLogger(__name__)
+
+
+__all__ = ["DatasetsActor", "MapDatasetActor"]
+
+
+class DatasetsActor(Datasets):
+    """A modified Dataset collection for parallel evaluation using ray actors.
+    Support only datasets composed of MapDataset.
+
+    Parameters
+    ----------
+    datasets : `Datasets`
+        Datasets
+    """
+
+    def __init__(self, datasets=None):
+        from ray import get
```

Comment:
I have a naive question: to my knowledge ray is not yet a mandatory dependency. Can you confirm?
If ray is not installed, this line will create a dirty error if a user has a code that resembles to the test function. Is it not better to raise a clean error in ray does not exist?
The `DatasetsActor`  is the main API interface, now....

Reply:
The message is `ModuleNotFoundError: No module named 'ray'` same as for the other optional modules (for example healpy), we don't have a special warning message.